screen-orientation/active-lock.html

```diff
@@ -4,22 +4,48 @@
 <script src="/resources/testdriver.js"></script>
 <script src="/resources/testdriver-vendor.js"></script>
 <p id="#fragment"></p>
-<script>
-  promise_test(async t => {
-    await test_driver.bless("request full screen", () => {
-      return document.documentElement.requestFullscreen();
-    });
-    const fragment = document.createElement("p");
-    fragment.id = "fragment";
-    document.body.appendChild(fragment);
-    const { type: preType } = screen.orientation;
-    const isPortrait = preType.startsWith("portrait");
-    const newType = `${isPortrait ? "landscape" : "portrait"}`;
-    const p = screen.orientation.lock(newType);
-    screen.orientation.onchange = t.unreached_func(
-      "change event must not fire"
-    );
+<iframe id="iframe"></iframe>
+<script type="module">
+  import { makeCleanup } from "./resources/orientation-utils.js";
+  promise_test(async (t) => {
+    t.add_cleanup(makeCleanup());
+    await test_driver.bless("request full screen");
+    await document.documentElement.requestFullscreen();
+    const orientation = screen.orientation.type.startsWith("portrait")
+      ? "landscape"
+      : "portrait";
+    const p = screen.orientation.lock(orientation);
     window.location.href = "#fragment";
     await p;
-  }, "When performing a fragment navigation, the orientation must not change or unlock");
+  }, "When performing a fragment navigation, the orientation change must not abort");
+
+  promise_test(async (t) => {
+    t.add_cleanup(makeCleanup());
+
+    // Get the iframe ready...
+    const iframe = document.getElementById("iframe");
+    await new Promise((resolve) => {
+      iframe.onload = resolve;
+      iframe.src = "about:blank";
+    });
+
+    // Keep references to things we need before navigating.
+    const iframeDoc = iframe.contentDocument;
+    const iframeWin = iframe.contentWindow;
+    const iframeException = iframeWin.DOMException;
+    const iframeOrientation = iframeWin.screen.orientation;
+
+    // start changing the orientation
+    await test_driver.bless("request full screen", null, iframeWin);
+    await iframeDoc.documentElement.requestFullscreen();
+    const orientation = iframeOrientation.type.startsWith("portrait")
+      ? "landscape"
+      : "portrait";
+    const p = iframeOrientation.lock(orientation);
+
+    // Navigate the iframe
+    iframe.src = "./resources/empty.html";
+
+    await promise_rejects_dom(t, "AbortError", iframeException, p);
```

Comment:
I slightly worry this is going to be racy. The navigation doesn't happen synchronously so it may not happen until after the promise has been resolved (successfully).

Reply:
Also, is this really per spec? The spec says:
"Whenever a [top-level browsing context](https://html.spec.whatwg.org/multipage/browsers.html#top-level-browsing-context) is [navigated](https://html.spec.whatwg.org/multipage/browsing-the-web.html#navigate), the [user agent](https://infra.spec.whatwg.org/#user-agent) MUST return the screen to the [default screen orientation](https://w3c.github.io/screen-orientation/#dfn-default-screen-orientation)."

It is however not a top-level browsing context that is being navigated here. Maybe we could do a similar test in a popup window instead of an iframe?